### DISCUSSION TITLE
pkgx.yaml syntax does not support @latest?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [pkgxdev/pkgx#891] [Help] [answered]
I was surprised to see `@latest` is not supported:

```yaml
# pkgx.yaml

dependencies:
  - ruby@latest
  - gem@latest
  - node@latest
  - npm@latest
  - python@latest
```

```bash
$ cd dotfiles
× nothing provides: npm@
│ we haven’t pkgd this yet. can you?
╰─➤ https://docs.pkgx.sh/pantry
```

If I remove the `@latest`, it loads at least, but I'm not sure if I always will get the _latest_ versions:

```bash
$ cd dotfiles
env +ruby-lang.org>=2.3 +nodejs.org +python.org +git-scm.org
```

Should I just be using the same syntax, but remove the `@latest`, to always get the latest versions?

**Comment by jhheider** [comment, marked as answer]
using `- ruby` or `ruby-lang.org: '*'` will install the latest version if no other version is on your machine; otherwise, it'll use the latest one on your machine if it satisfies all other constraints.

this is somewhat by design (though @latest should probably be supported). dev dependencies should usually be fairly well-defined.

**Reply by fredrikaverpil** [reply]
Ok, great - thanks for confirming.

> dev dependencies should usually be fairly well-defined

Yeah, I actually use this only so that I can `cd` into my dotfiles and upgrade all plugins and tooling in Neovim from that location. It needs this tooling, and I like to always use the latest stuff in there.
